clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -79,39 +81,35 @@ export class HttpConnection implements IConnection {
                 // No need to add a connection ID in this case
                 this.url = this.baseUrl;
                 this.transport = this.constructTransport(TransportType.WebSockets);
+                // We should just call connect directly in this case.
+                // No fallback or negotiate in this case.
+                await this.transport.connect(this.url, transferFormat, this);
             } else {
                 let headers;
                 const token = this.options.accessTokenFactory();
                 if (token) {
                     headers = {
```

Comment:
You can get rid of this variable and just assign directly to `this.headers` and use that.

Reply:
Yeah. Was hitting scoping issues before. I'll clean this up